sdk/analytics/src/main/java/com/klaviyo/analytics/networking/KlaviyoApiClient.kt

```diff
@@ -96,24 +131,35 @@ internal object KlaviyoApiClient : ApiClient {
         // Keep track if there's any errors restoring from persistent store
         var wasMutated = false
 
+        Registry.log.info("Restoring persisted queue")
+
         Registry.dataStore.fetch(QUEUE_KEY)?.let {
             try {
                 val queue = JSONArray(it)
                 Array(queue.length()) { i -> queue.optString(i) }
             } catch (exception: JSONException) {
                 wasMutated = true
+                Registry.log.wtf("Invalid persistent queue JSON", exception)
+                Registry.log.debug(it)
                 emptyArray<String>()
             }
         }?.forEach { uuid ->
-            Registry.dataStore.fetch(uuid)?.let { json ->
-                try {
-                    val request = KlaviyoApiRequest.fromJson(JSONObject(json))
-                    if (!apiQueue.contains(request)) {
-                        apiQueue.offer(request)
-                    }
-                } catch (exception: JSONException) {
+            Registry.dataStore.fetch(uuid).let { json ->
+                if (json == null) {
+                    Registry.log.error("Missing request JSON for $uuid")
                     wasMutated = true
-                    Registry.dataStore.clear(uuid)
+                } else {
+                    try {
+                        val request = KlaviyoApiRequest.fromJson(JSONObject(json))
+                        if (!apiQueue.contains(request)) {
+                            apiQueue.offer(request)
+                        }
+                    } catch (exception: JSONException) {
+                        wasMutated = true
+                        Registry.log.wtf("Invalid request JSON $uuid", exception)
```

Comment:
Nice I like how you add wtf!

Reply:
i mean how could i not once i saw it in android logger...